aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsFluentClientDecorator.kt

```diff
@@ -177,21 +178,17 @@ private class AwsFluentClientExtensions(types: Types) {
                 ##[cfg(any(feature = "rustls", feature = "native-tls"))]
                 pub fn from_conf(conf: crate::Config) -> Self {
                     let retry_config = conf.retry_config().cloned().unwrap_or_else(#{RetryConfig}::disabled);
-                    let timeout_config = conf.timeout_config().cloned().unwrap_or_default();
+                    let timeout_config = conf.timeout_config().cloned().unwrap_or_else(#{TimeoutConfig}::disabled);
                     let sleep_impl = conf.sleep_impl();
                     if (retry_config.has_retry() || timeout_config.has_timeouts()) && sleep_impl.is_none() {
                         panic!("An async sleep implementation is required for retries or timeouts to work. \
                                 Set the `sleep_impl` on the Config passed into this function to fix this panic.");
                     }
                     let mut builder = #{aws_smithy_client}::Builder::dyn_https()
-                        .middleware(#{DynMiddleware}::new(#{Middleware}::new()));
-                    builder.set_retry_config(retry_config.into());
-                    builder.set_timeout_config(timeout_config);
-                    // the builder maintains a try-state. To avoid suppressing the warning when sleep is unset,
-                    // only set it if we actually have a sleep impl.
-                    if let Some(sleep_impl) = sleep_impl {
-                        builder.set_sleep_impl(Some(sleep_impl));
-                    }
+                        .middleware(#{DynMiddleware}::new(#{Middleware}::new()))
+                        .retry_config(retry_config.into())
+                        .timeout_config(timeout_config);
```

Comment:
Note for myself: I started integration testing the connect timeout, and that revealed that the `timeout_config` here gets discarded since `dyn_https()` creates the connector and has no consideration of the `timeout_config` value at all.

This is another one of those weird situations with the `aws_smithy_client::Builder` similar to `retry_config` where you can configure a default implementation, or you can replace that implementation wholesale. I think we need to redesign this builder so that these conflicting builder methods are no longer possible. If you replace an implementation, then it shouldn't be possible to set configuration for the default.